app/code/Magento/Customer/Model/Address.php

```diff
@@ -193,7 +193,7 @@ public function getDataModel($defaultBillingAddressId = null, $defaultShippingAd
      */
     public function getCustomerId()
     {
-        return $this->_getData('customer_id') ? $this->_getData('customer_id') : $this->getParentId();
+        return $this->_getData('customer_id') ? (int) $this->_getData('customer_id') : (int) $this->getParentId();
```

Comment:
In case if you'll create new model and will egecute this method, like this:
```php
$address = $this->addressFactory->create();
$address->getCustomerId();
```

Result will be like this: `(int) $this->getParentId();`  => `(int)null`  => `0`. I believe this is NOT expected result, it should be null.